components/renku_data_services/project/api.spec.yaml

```diff
@@ -0,0 +1,382 @@
+        created_by:
+          $ref: "#/components/schemas/User"
+        repositories:
+          $ref: "#/components/schemas/RepositoriesList"
+        visibility:
+          $ref: "#/components/schemas/Visibility"
+        members:
+          $ref: "#/components/schemas/MembersWithRoles"
+      required:
+        - "id"
+        - "name"
+        - "slug"
+        - "created_by"
+
+    Ulid:
+      description: ULID identifier
+      type: string
+      minLength: 26
+      maxLength: 26
+      pattern: "^[A-Za-z0-9]{26}$"
+      format: ulid
+    Name:
+      description: Renku project name
+      type: string
+      minLength: 1
+      maxLength: 99
+      example: "My Renku Project :)"
+    Slug:
+      description: A command-line friendly name for a project
+      type: string
+      minLength: 1
+      maxLength: 99
+      pattern: "^[a-z0-9]+[a-z0-9._-]*$"  # Cannot contain consecutive special characters or end in .git and .atom
+      example: "my-renku-project"
+    CreationDate:
+      description: The date and time the project was created
+      type: string
+      format: date-time
+      example: "2023-11-01T17:32:28Z"
+    User:
+      description: A KeyCloak user
+      type: object
+      additionalProperties: false
+      properties:
+        id:
+          $ref: "#/components/schemas/UserId"
+      required:
+        - "id"
+      example:
+        id: "some-keycloak-user-id"
+    UserId:
+      description: KeyCloak user ID
+      type: string
+      pattern: "^[A-Za-z0-9-]+$"
+      minLength: 1
+      example: "123-keycloak-user-id-456"
+    RepositoriesList:
+      description: A list of repository URLs
+      type: array
+      items:
+        $ref: "#/components/schemas/GitUrl"
+      minItems: 0
+      uniqueItems: true
+    GitUrl:
+      description: URL of a Git repository
+      type: string
+      format: uri  # TODO: This should support other formats like ``git@`` and ``https+git``
+    Visibility:
+      description: Project's visibility levels
+      type: string
+      enum:
+        - "public"
+        - "private"
+    MembersWithRoles:
+      description: List of users and their access level to the project
+      type: array
+      items:
+        $ref: "#/components/schemas/MemberWithRole"
+      minItems: 0
+      uniqueItems: true
+    MemberWithRoles:
+      description: A user and the access level to the project
+      type: object
+      additionalProperties: false
+      properties:
+        user:
+          $ref: "#/components/schemas/User"
+        role:
+          $ref: "#/components/schemas/Role"
+      required:
+        - user
+        - role
+    Role:
+      # TODO: Can have multiple owners? Owner here vs ``created_by``?
+      # TODO: Better names for roles: Maintainer, admin, member, ... instead of reader and writer
+      description: Possible roles of users in a project
+      type: string
+      enum:
+        - "reader"
+        - "writer"
+        - "owner"
```

Comment:
Lets just follow what the notion page says. We have `owner`, `member` and `admin` roles. 

It is this secition in notion:
<img width="724" alt="image" src="https://github.com/SwissDataScienceCenter/renku-data-services/assets/16360283/4e3ead45-4026-4dd1-b5b3-df5c50f9eee6">


The `created_by` field is the id of the user who created the project. That should not be changeable. And it could be that the original creator is not in the project members, owners, etc at all anymore.